### ISSUE TITLE
Setting hook.result to NULL returns 404

### ISSUE THREAD

**Issue by zender** [feathersjs/feathers#391]
When I try to set the result to null in after hook of remove method then I get 404 instead of 204:

remove: [
    (hook, next) => {
      hook.result = null;
      next(null, hook);
    }
  ]

It will be useful if after hook.result is set to null then No Content (204) to be returned.

**Comment by daffl** [comment]
feathers-rest already handles 204 [here](https://github.com/feathersjs/feathers-rest/blob/master/src/wrappers.js#L62) and setting `hook.result` to anything other than `undefined` is handled [here](https://github.com/feathersjs/feathers-hooks/blob/master/src/hooks.js#L42).
What does the rest of your setup look like?

**Comment by daffl** [comment]
I just tried the following example application:

```js
const feathers = require('feathers');
const rest = require('feathers-rest');
const hooks = require('feathers-hooks');
const bodyParser = require('body-parser');
const handler = require('feathers-errors/handler');

const app = feathers();

app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: true }));
app.configure(hooks());
app.configure(rest());
app.use(handler());
app.use('/todos', {
  get() {
    return Promise.resolve({
      test: 'me'
    });
  }
});


app.service('todos').before({
  get(hook) {
    hook.result = null;
  }
});

app.listen(3030);
```

And `GET todos/anything` does return a 204 status code.

**Comment by zender** [comment]
Thank you for the quick reply.  It seems that it does not work with mongosee adapter. Here you can find some sample code: https://github.com/zender/feathers-204.

I tried it with before and after hook (patch, update and remove) - return 404 and message "Can't set headers after they are sent.
"

**Comment by daffl** [comment]
Perfect, thank you for the example repo. Made it really easy to track down. I have a fix in https://github.com/feathersjs/feathers-rest/pull/67 which should be released shortly.

**Comment by daffl** [comment]
Fixed in `feathers-rest` v1.4.4
